### PR TITLE
core: split notarypool and mempool metrics

### LINKED ISSUE #2950: Split mempool/notary pool Prometheus metrics
The only metric we have for mempool is `mempool_unsorted_tx`. However, mempool is used both for simple transactions and for notary requests. As a result, we've got a mess in the output, we have to keep two separate metrics for these pools.

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Also, do not use metrics for temporary pools created during block/transaction verification.

Close #2950.

### CODE REVIEW

File: pkg/core/mempool/prometheus.go [hunk 1]
```diff
@@ -11,14 +11,29 @@ var (
...
 func init() {
 	prometheus.MustRegister(
 		mempoolUnsortedTx,
+		notarypoolUnsortedTx,
 	)
 }
 
-func updateMempoolMetrics(unsortedTxnLen int) {
+// UpdateMempoolMetrics updates metric of the number of unsorted txs inside the mempool.
+func UpdateMempoolMetrics(unsortedTxnLen int) {
```
Comment (roman-khimov) [inline pkg/core/mempool/prometheus.go:32]: Maybe we can move these metrics out of `mempool` into `core` and `network` respectively?

Comment (AnnaShaleva) [reply pkg/core/mempool/prometheus.go]: Yes, looks better this way, fixed.

File: pkg/core/prometheus.go [hunk 2]
```diff
@@ -6,37 +6,46 @@ import (
...
 			Help:      "Current header height",
 			Name:      "current_header_height",
 			Namespace: "neogo",
 		},
 	)
+	// mempoolUnsortedTx prometheus metric.
+	mempoolUnsortedTx = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Help:      "Mempool unsorted transactions",
+			Name:      "mempool_unsorted_tx",
```
Comment (AnnaShaleva) [inline pkg/core/prometheus.go:37]: @roman-khimov, I've changed the name of this metric. I doubt anyone uses it in some scripts, but still, do you have any objections?

Comment (roman-khimov) [reply pkg/core/prometheus.go:37]: IIUC you've changed the `help` string, but the `name` is the same. It's OK this way. Changing names is not OK, that's a part of the public node interface.

**Review by roman-khimov — APPROVED** [review]

File: pkg/core/prometheus.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/core/prometheus.go:37]: Oh, you're right, somehow I thought that I've changed the name as far. So it's just a help, yes.

**Pull request merged**